core/src/main/java/com/datastrato/gravitino/catalog/TableNormalizeDispatcher.java

```diff
@@ -82,9 +82,7 @@ public Table alterTable(NameIdentifier ident, TableChange... changes)
 
   @Override
   public boolean dropTable(NameIdentifier ident) {
-    // The constraints of the name spec may be more strict than underlying catalog,
-    // and for compatibility reasons, we only apply case-sensitive capabilities here.
-    return dispatcher.dropTable(applyCaseSensitive(ident, Capability.Scope.TABLE, dispatcher));
+    return dispatcher.dropTable(normalizeNameIdentifier(ident));
```

Comment:
after these changes, can I drop a Mysql table named `a&b` by Gravitino?

Reply:
No, as the current regex pattern in this PR does not try to match the character `&`.
Should we add `&` to the regex pattern?

Reply:
No, this is just my assumption. 

We can gradually relax the strategy after encountering real needs, or make this rule configurable when the demand is frequent.